### ISSUE TITLE
Deserialization problem on Android

### ISSUE THREAD

**Issue by altugygt** [deeplearning4j/deeplearning4j#3170]
Hi, I saved my network as zip file. I able to deserialize file through Maven java project but when I try that on Android project I'm getting runtime exception:

"D/Error: ERR: stack=java.lang.RuntimeException: Unable to start activity java.lang.RuntimeException: org.nd4j.shade.jackson.databind.JsonMappingException: Could not resolve type id 'dropout' into a subtype of [simple type, class org.deeplearning4j.nn.conf.layers.Layer]: known type ids = [Layer, RBM, VariationalAutoencoder, activation, autoEncoder, batchNormalization, convolution, dense, embedding, gravesBidirectionalLSTM, gravesLSTM, localResponseNormalization, loss, output, rnnoutput, subsampling]"

Model JSON: {
  "iterationCount" : 0,
  "l1ByParam" : { },
  "l2ByParam" : { },
  "layer" : {
    "convolution" : {
      "activationFn" : {
        "Identity" : { }
      },
      "adamMeanDecay" : "NaN",
      "adamVarDecay" : "NaN",
      "biasInit" : 0.0,
      "biasLearningRate" : 0.1,
      "convolutionMode" : "Truncate",
      "cudnnAlgoMode" : "PREFER_FASTEST",
      "dist" : null,
      "dropOut" : 1.0,
      "epsilon" : "NaN",
      "gradientNormalization" : "None",
      "gradientNormalizationThreshold" : 1.0,
      "kernelSize" : [ 5, 16 ],
      "l1" : 0.0,
      "l1Bias" : 0.0,
      "l2" : 0.0,
      "l2Bias" : 0.0,
      "layerName" : "convolution2d_1",
      "learningRate" : 0.1,
      "learningRateSchedule" : null,
      "momentum" : "NaN",
      "momentumSchedule" : null,
      "nin" : 3,
      "nout" : 16,
      "padding" : [ 0, 0 ],
      "rho" : "NaN",
      "rmsDecay" : "NaN",
      "stride" : [ 1, 1 ],
      "updater" : "SGD",
      "weightInit" : "XAVIER_UNIFORM"
    }
  },
  "leakyreluAlpha" : 0.0,
  "learningRateByParam" : { },
  "learningRatePolicy" : "None",
  "lrPolicyDecayRate" : "NaN",
  "lrPolicyPower" : "NaN",
  "lrPolicySteps" : "NaN",
  "maxNumLineSearchIterations" : 5,
  "miniBatch" : true,
  "minimize" : true,
  "numIterations" : 1,
  "optimizationAlgo" : "STOCHASTIC_GRADIENT_DESCENT",
  "pretrain" : false,
  "seed" : 1490995055667,
  "stepFunction" : null,
  "useDropConnect" : false,
  "useRegularization" : false,
  "variables" : [ "0_W", "0_b", "2_W", "2_b", "6_W", "6_b", "9_W", "9_b" ]
}

I'm using restoreMultiLayerNetwork() method to import model. Should I not use dropout on Android?

**Comment by agibsonccc** [comment]
Could you upload a copy of the model for us to look at? Are you on the latest version? Neural network features aren't "limited" to a particular OS or anything. That would be silly.

**Comment by altugygt** [comment]
I'm using 0.8.0 version. I'm not able to import Keras model directly on Android because HDF5 isn't available for Android. I've imported network via importKerasSequentialModelAndWeights() method on Maven DL4J project then saved model as zip file. I'm trying deserialize this model file on Android. As shown below:

`MultiLayerNetwork model = KerasModelImport.importKerasSequentialModelAndWeights(kerashdf5_path);`
`File file = new File("...\KerasDl4jMln.zip");`
`ModelSerializer.writeModel(model, file, true);`

`MultiLayerNetwork restoredModel = ModelSerializer.restoreMultiLayerNetwork(file) ;// Android`

Model is here: https://drive.google.com/open?id=0BwYg9SwaRaDWenMyclhYOUdGdTQ

**Comment by maxpumperla** [comment]
@altugygt I'm taking care of this issue now. Do you still have problems with this? Sorry it took so long to get back to you. Thanks.

**Comment by maxpumperla** [comment]
@altugygt do you still see the same problems? We fixed a lot in the last months, you're likely fine with the latest snapshots. Will close for now, please reopen if your problems persist. thanks
